_posts/2016-03-16-code-smells-part-I.md

```diff
@@ -0,0 +1,259 @@
+
+
+To do this blog I started to look at my old code when I hadn't woken up yet to craftsmanship: If it was working that was good enough for me. Here's the code in Objective-C:
+
+`````
+- (void) postToServer
+{
+    PostSerializer* postSerializer = [[PostSerializer alloc] init];
+    NSString *post = [postSerializer serializePostWithTitle:self.txtTitle.text description:self.txtDescription.text author:self.txtUser.text game:self.game];
+
+    NSMutableDictionary *postParams = [NSMutableDictionary dictionary];
+	[postParams setObject:txtTitle.text forKey:@"title"];
+	[postParams setObject:post forKey:@"data"];
+	[postParams setObject:txtUser.text forKey:@"username"];
+    [postParams setObject:txtPassword.text forKey:@"password"];
+
+	NSArray *args = [NSArray arrayWithObjects:[NSNumber numberWithInt:0], postParams, nil];
+
+#ifdef DEBUG_LOG
+    XMLRPCRequest *request = [[XMLRPCRequest alloc] initWithURL:
+                              [NSURL URLWithString:@"http://localhost:8888/letsbasket/xmlrpc.php"]];
+    DLog(@"Debug");
+#else
+    XMLRPCRequest *request = [[XMLRPCRequest alloc] initWithURL:[NSURL URLWithString:[UtilsHelper localizeString:@"UrlXmlRPCKey"]]];
+    DLog(@"Producao");
+#endif
+
+	[request setMethod:@"letsBasket.AddPost" withParameters:args];
+
+    NSError *error = nil;
+	XMLRPCResponse* result = [XMLRPCConnection sendSynchronousXMLRPCRequest:request error:&error];
+
+    //DLog(@"%@", [request body]);
+    //DLog(@"%@", [result body]);
+
+    UIApplication *app = [UIApplication sharedApplication];
+    app.networkActivityIndicatorVisible = NO;
+
+    [self dismissWaitingAlert];
+
+    if(error != nil || [[result body] rangeOfString:@"<name>error</name>"].location != NSNotFound)
+    {
+        int location_start = [[result body] rangeOfString:@"<string>"].location + 8;
+        int location_end = [[result body] rangeOfString:@"</string>"].location;
+
+        NSString *message = [[[result body] substringWithRange:NSMakeRange(location_start, location_end- location_start)] unescapedString];
+        NSString* title = [UtilsHelper localizeString:@"PublishVC_ErrorRetreivingAlertTitle_key"];
+        [self showAlertWithErrorMessage:message Title:title];
+	return;
+	}
+
+    [self processPublishResult:result];
+}
+`````
+
+Wow! This is really a big method. And this is inside a ViewController class, so definitely this should be extracted into a service class, so we have a correct separation of concerns. But for the sake of the brevity, let's focus on how can we refactor this big method.
```

Comment:
Wow! This is a really big method. And it is inside a ViewController class, so this should definitely be extracted into a service class, so we have a correct separation of concerns. But for the sake of the brevity, let's focus on how can we refactor this big method.